### PR TITLE
Use new action

### PR DESCRIPTION
**Pull request by olgarithms**

### CODE REVIEW

File: docs/10-automating-versioning.md [hunk 1]
```diff
@@ -24,11 +24,20 @@ on:
...
-Next, modify the step `Install dependencies` to install `sphinx_multiversion`:
+Since we need to display the content of other git branches and tags, we need to have all history available. On the step of checking-out our repo, configure the `Checkout V3` action to fetch all history for all branches and tags:
+
+```yaml
+- uses: actions/checkout@v3
+  with:
+    fetch-depth: 0
+```
+
+As expected, you need to modify install `sphinx_multiversion` in `Install dependencies` step:
```
Comment (aelsayed95) [inline docs/10-automating-versioning.md]: modify install `sphinx_multiversion`?

File: docs/10-automating-versioning.md [hunk 2]
```diff
@@ -38,48 +47,44 @@ documentation:
...
-          git commit -m "deploy documentation updates"
-          git push origin gh-pages
+          sphinx-multiversion docs docs/_build/html
+      - name: Deploy
+        uses: peaceiris/actions-gh-pages@v3
+        with:
+          publish_branch: gh-pages
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          publish_dir: docs/_build/html
+          force_orphan: true
```
Comment (aelsayed95) [inline docs/10-automating-versioning.md:87]: nice

File: docs/8-automating-documentation-updates.md [hunk 3]
```diff
@@ -73,98 +73,100 @@ on:
...
-          make html
-      - name: Deploy docs
-        run: |
-          cd docs/_build/html
-          git add --all
-          git commit -m "deploy documentation updates"
-          git push origin gh-pages
+- name: Sphinx build
+  run: |
+    sphinx-apidoc -o docs src/sphinxy/ --separate --force
```
Comment (aelsayed95) [inline docs/8-automating-documentation-updates.md]: is --force needed?

File: docs/8-automating-documentation-updates.md [hunk 4]
```diff
@@ -73,98 +73,100 @@ on:
...
+    sphinx-build docs docs/_build/html
```
Comment (aelsayed95) [inline docs/8-automating-documentation-updates.md:117]: idk if people are familiar with this cmd cuz we've been using `make html` should we explain on L111?

File: docs/8-automating-documentation-updates.md (hunk 3 above)
Comment (olgarithms) [reply docs/8-automating-documentation-updates.md]: Probably not

File: docs/8-automating-documentation-updates.md (hunk 4 above)
Comment (olgarithms) [reply docs/8-automating-documentation-updates.md:117]: We have explained it here: https://olgarithms.github.io/sphinx-tutorial/docs/2-getting-started-with-sphinx.html

> Executing make html runs sphinx-build, pointing to the source and build directories to use and instructs Sphinx to build html pages out of your rst files.

**Review by aelsayed95 — APPROVED** [review]

**Pull request merged**